addons/website/static/src/snippets/s_searchbar/000.js

```diff
@@ -222,6 +224,22 @@ publicWidget.registry.searchBar = publicWidget.Widget.extend({
         if ($prevMenu) {
             $prevMenu.remove();
         }
+        // Adjust the menu's position based on the scroll height.
+        if (res && this.limit) {
+            this.el.classList.remove("dropup");
+            const wrapwrapEl = document.querySelector("#wrapwrap");
+            if (wrapwrapEl.scrollHeight > pageScrollHeight) {
+                // If the menu overflows below the page, we reduce its height.
+                this.$menu[0].style.maxHeight = "40vh";
+                this.$menu[0].style.overflowY = "auto";
+                // We then recheck if the menu still overflows below the page.
+                if (wrapwrapEl.scrollHeight > pageScrollHeight) {
+                    // If the menu still overflows below the page after its height
+                    // has been reduced, we position it above the input.
+                    this.el.classList.add("dropup");
+                }
```

Comment:
Do we still want the reduce height if we do a dropup?

Reply:
> Do we still want the reduce height if we do a dropup?

It could be, but I dont want to redo the check once the dropdown is up ...

=> check if enough place down
=> if not, reduce it
=> if still not, put up
=> check if enough place up
=> if not, reduce it

I could do that but I thought it was overkill, I don't know maybe it is a question for PDE